index.js

```diff
@@ -1,10 +1,18 @@
 'use strict';
 
+// Polyfill Promise if running on Node < 0.11
+// Note:  Must modify global Promise for buffered-spawn
+if ('undefined' === typeof Promise) {
+  global.Promise = require('bluebird').Promise;
+}
+
 var spawn = require('cross-spawn')
   , which = require('which')
   , path = require('path')
   , util = require('util')
-  , tty = require('tty');
+  , tty = require('tty')
+  , bufferedSpawn = require('buffered-spawn')
```

Comment:
The `buffered-spawn` isn't really needed. As this a CLI file, there really is no need for async execution. That's why we're using an `sync` exec throughout the code base. Using https://github.com/observing/pre-commit/blob/master/index.js#L67-L71 would make the code a lot easier to read manage as there would be no need for promises and async flow control.